src/textCampaign/services/textCampaign.service.ts

```diff
@@ -0,0 +1,243 @@
+import {
+  forwardRef,
+  Inject,
+  Injectable,
+  Logger,
+  NotFoundException,
+} from '@nestjs/common'
+import { PrismaService } from 'src/prisma/prisma.service'
+import { CreateProjectSchema } from '../schemas/createProject.schema'
+import { RumbleUpService } from './rumbleUp.service'
+import { CampaignsService } from '../../campaigns/services/campaigns.service'
+import { TextCampaignSummary } from '../textCampaign.types'
+import { TextCampaignStatus } from '@prisma/client'
+
+@Injectable()
+export class TextCampaignService {
+  public readonly logger = new Logger(TextCampaignService.name)
+
+  constructor(
+    @Inject(forwardRef(() => CampaignsService))
+    private readonly campaignsService: CampaignsService,
```

Comment:
This really feels like a code-smell here.

We're injecting the entire `CampaignService` here, cyclically even, just to do what amounts to determining if a given `campaignId` actually exists, right (down in `createProject`, and _maybe_ `uploadContactsAndCreateProject` if that dead code isn't removed for the time being).

Seems like there could be a better ACL-type solution here?

Maybe we need to create a custom NestJS "guard" to make this determination up in the controller, as opposed to down here in the service? 🤔